pkg/config/config.go

```diff
@@ -252,7 +260,7 @@ func fillOpenShiftRouteSuffixIfNecessary(nonCachedClient client.Client, configMa
 	host := testRoute.Spec.Host
 	if host != "" {
 		prefixToRemove := "che-workspace-controller-test-route-" + configMap.Namespace + "."
```

Comment:
It's legacy functionality I didn't even know existed until I worked on this PR; I'm inclined to remove it entirely if there's doubts as to its usefulness. Thus far, the concerns are
- It depends on deployed namespace being `che-workspace-controller`, will fail if we change that
- Will fail if how default hostnames are computed is changed
- Only works on OpenShift

This kind of points towards "we shouldn't support this at the moment". Executing this only when the entry in the cm is blank means that we just move all those failure cases down the line, except it makes it look like we're changing config requirements rather than the cluster is changing how it resolves things.

Regarding 
>Also, che-workspace-controller-test-route-che-workspace-controller is near to drain 64 chars limit ) Consider making test route name shorter.

This is actually not a problem; OpenShift will generously generate an invalid hostname and only tell you its a problem later:
```
$ oc get route che-plugin-registry-abcdefghijklmnopqrstuvwxyz -o yaml | grep host:
  host: che-plugin-registry-abcdefghijklmnopqrstuvwxyz-che-workspace-controller.apps-crc.testing
      message: 'host name validation errors: spec.host: Invalid value: "che-plugin-registry-abcdefghijklmnopqrstuvwxyz-che-workspace-controller.apps-crc.testing":
    host: che-plugin-registry-abcdefghijklmnopqrstuvwxyz-che-workspace-controller.apps-crc.testing
```

However, it might make sense to try and use `.status.ingress[].routerCanonicalHostname`:
```
$ oc get route che-plugin-registry-abcdefghijklmnopqrstuvwxyz -o yaml | yq '.status.ingress[].routerCanonicalHostname'
"apps-crc.testing"
```

Reply:
> This is actually not a problem; OpenShift will generously generate an invalid hostname and only tell you its a problem later :) :

:-D It's true and it may be strange but it helps use in this case

Reply:
Looked into using the status field, but it's not populated immediately. For now I'm leaving the functionality in place, but we should test its reliability in the future and potentially remove/improve it.